packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -46,9 +50,18 @@ export default function astro({ settings, logger }: AstroPluginOptions): vite.Pl
 			}
 			// For CSS / hoisted scripts, the main Astro module should already be cached
 			const filename = normalizePath(normalizeFilename(parsedId.filename, config.root));
-			const compileResult = getCachedCompileResult(config, filename);
+			let compileResult = getCachedCompileResult(config, filename);
 			if (!compileResult) {
-				return null;
+				// In dev, HMR could cause this compile result to be empty, try to load it first
+				if (server) {
+					await server.transformRequest('/@fs' + filename);
```

Comment:
this is inside load, this doesn't cause a loop?

Reply:
The `id` of this `load` is something like `/Users/bjorn/project/src/pages/index.astro/?astro&lang.css`, but the filename here would be `/Users/bjorn/project/src/pages/index.astro`. So it won't cause a loop and should only transform the main Astro file so we have the cache ready.